source/MaterialXTest/ShaderValid.cpp

```diff
@@ -302,33 +302,31 @@ static mx::GlslValidatorPtr createValidator(bool& orthographicView, const std::s
 //
 // Outputs error log if validation fails
 //
-static void runValidation(const std::string& shaderName, std::vector<mx::ElementPtr> elementList,
+static void runValidation(const std::string& outputPath, const std::string& shaderName, mx::ElementPtr element,
                           mx::GlslValidatorPtr validator, mx::ShaderGeneratorPtr shaderGenerator,
                           bool orthographicView, mx::DocumentPtr doc, std::ostream& log)
 {
     mx::SgOptions options;
 
-    for (auto elemPtr : elementList)
+    if(element && doc)
     {
-        if (!elemPtr)
-        {
-            continue;
-        }
-        log << "------------ Run validation with element: " << elemPtr->getName() << std::endl;
+        log << "------------ Run validation with element: " << element->getName() << std::endl;
 
-        mx::ShaderPtr shader = shaderGenerator->generate(shaderName, elemPtr, options);
+        std::string shaderPath =  outputPath + "/" + shaderName;
+        mx::ShaderPtr shader = shaderGenerator->generate(shaderName, element, options);
         mx::HwShaderPtr hwShader = std::dynamic_pointer_cast<mx::HwShader>(shader);
         REQUIRE(hwShader != nullptr);
         REQUIRE(hwShader->getSourceCode(mx::HwShader::PIXEL_STAGE).length() > 0);
         REQUIRE(hwShader->getSourceCode(mx::HwShader::VERTEX_STAGE).length() > 0);
 
-        mx::writeToXmlFile(doc, elemPtr->getName() + ".mtlx");
+        // TODO: Put this back in when we are writing to a results folder, as we'll want a copy of the document
```

Comment:
As the element may be part of something procedurally generated. We may want this as an option.